docs/headerparser_usage.md

```diff
@@ -0,0 +1,77 @@
+## Headerparser
+
+headerparser is a tool that assists in writing device system call descriptions for syzkaller.
+
+In order to make Syzkaller smarter when it comes to fuzzing a device node, you can provide it with
+information about the ioctl argument struct types it expects.
+
+However, in certain cases the number of argument struct types might be high, increasing the amount of manual
+effort that goes into writing the description files for the struct types.
+
+In order to ease the effort of writing ioctl argument type description files, headerlib does a best-effort job at generating them for you. You will still need to manually select the appropriate Syzkaller data type from the list of types [here](https://github.com/google/syzkaller/blob/master/docs/syscall_descriptions_syntax.md).
+
+## Dependencies
+Headerlib uses pycparser. Please make sure you install pycparser as described below. If not you might run into a [pycparser bug](https://github.com/pyca/cryptography/issues/3187).
+
+```shell
+$ pip install --no-binary pycparser
+```
+
+## Using headerparser
+```shell
+$ python headerparser.py --device th_b --filenames=./test_headers/th_b.h
+-=============== Structures -]
+
+-=============== Structure Metadata -]
+B {
+                            B1   len|fileoff|flags|intN
+                            B2   len|fileoff|flags|intN
+}
+struct_containing_union {
+                     something   len|fileoff|flags|intN
+                 a_uion.a_char   string|filename
+                  a_uion.B_ptr   ptr|buffer|array
+}
+```
+
+You can copy paste the content underneath the `Structure Metadata` over to your syzkaller device description.
+
+## Something breaks
+Let us try parsing `test_headers/th_a.h` header file to generate argument structs.
+
+```shell
+$ python headerparser.py --device th_a --filenames=./test_headers/th_a.h
+Traceback (most recent call last):
```

Comment:
Can we please have for syntax errors just:
```
pycparser.plyparser.ParseError: /tmp/tmpNtK6CQ/source.o:34:2: before: bool
```
instead of:
```
Traceback (most recent call last):
  File "tools/syz-headerparser/headerparser.py", line 65, in <module>
    main()
  File "tools/syz-headerparser/headerparser.py", line 57, in main
    output_fmt=args.outputfmt)
  File "/homegopath/src/github.com/google/syzkaller/tools/syz-headerparser/headerlib/container.py", line 167, in __init__
    self.load_header_files()
  File "/homegopath/src/github.com/google/syzkaller/tools/syz-headerparser/headerlib/container.py", line 199, in load_header_files
    loglvl=self.loglvl)
  File "/homegopath/src/github.com/google/syzkaller/tools/syz-headerparser/headerlib/struct_walker.py", line 126, in __init__
    loglvl=loglvl).get_ast()
  File "/homegopath/src/github.com/google/syzkaller/tools/syz-headerparser/headerlib/header_preprocessor.py", line 137, in get_ast
    return self._get_ast()
  File "/homegopath/src/github.com/google/syzkaller/tools/syz-headerparser/headerlib/header_preprocessor.py", line 134, in _get_ast
    return pycparser.parse_file(self.temp_objectfile)
  File "/usr/local/lib/python2.7/dist-packages/pycparser/__init__.py", line 93, in parse_file
    return parser.parse(text, filename)
  File "/usr/local/lib/python2.7/dist-packages/pycparser/c_parser.py", line 152, in parse
    debug=debuglevel)
  File "/usr/local/lib/python2.7/dist-packages/pycparser/ply/yacc.py", line 331, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "/usr/local/lib/python2.7/dist-packages/pycparser/ply/yacc.py", line 1199, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
  File "/usr/local/lib/python2.7/dist-packages/pycparser/ply/yacc.py", line 193, in call_errorfunc
    r = errorfunc(token)
  File "/usr/local/lib/python2.7/dist-packages/pycparser/c_parser.py", line 1761, in p_error
    column=self.clex.find_tok_column(p)))
  File "/usr/local/lib/python2.7/dist-packages/pycparser/plyparser.py", line 66, in _parse_error
    raise ParseError("%s: %s" % (coord, msg))
pycparser.plyparser.ParseError: /tmp/tmpNtK6CQ/source.o:34:2: before: bool
```

This is not useful and unreadable.